File: IronNails.Contracts/Models/DirectMessage.cs
namespace IronNails.Models
{
    public class DirectMessage : TweetBase
    {
        #region Fields

        private string _id;
        private User _recipient;
        private string _recipientId;
        private string _recipientScreenName;
        private User _sender;
        private string _senderId;
        private string _senderScreenName;
        private string _text;

        #endregion

        #region Properties

        public string Id
        {
            get { return _id; }
            set
            {
                if (value == _id) return;
                _id = value;
                OnPropertyChanged("MessageId");
            }
        }

        public string Text
        {
            get { return _text; }
            set
            {
                if (value == _text) return;
                _text = value;
                OnPropertyChanged("Text");
            }
        }

        public string SenderId
        {
            get { return _senderId; }
            set
            {
                if (value == _senderId) return;
                _senderId = value;
                OnPropertyChanged("SenderId");
            }
        }

        public string RecipientId
        {
            get { return _recipientId; }
            set
            {
                if (value == _recipientId) return;
                _recipientId = value;
                OnPropertyChanged("RecipientId");
            }
        }

        public string SenderScreenName
        {
            get { return _senderScreenName; }
            set
            {
                if (value == _senderScreenName) return;
                _senderScreenName = value;
                OnPropertyChanged("SenderScreenName");
            }
        }

        public string RecipientScreenName
        {
            get { return _recipientScreenName; }
            set
            {
                if (value == _recipientScreenName) return;
                _recipientScreenName = value;
                OnPropertyChanged("RecipientScreenName");
            }
        }

        public User Sender
        {
            get { return _sender; }
            set
            {
                if (value == _sender) return;
                _sender = value;
                OnPropertyChanged("Sender");
            }
        }

        public User Recipient
        {
            get { return _recipient; }
            set
            {
                if (value == _recipient) return;
                _recipient = value;
                OnPropertyChanged("Recipient");
            }
        }

        #endregion
    }
}

File: IronNails/app/models/direct_message.rb
module IronNails

  module Models

    class DirectMessage

      @@properties = %w(created_at id text sender_id recipient_id sender_screen_name recipient_screen_name)
      @@children = %w(sender recipient)

      def self.properties
        @@properties
      end

      def self.children
        @@children
      end

      attr_accessor *properties
      attr_accessor *children

      def self.all(credentials, &callback)
        options = { :url => Constants::Urls::DIRECT_MESSAGES, :credentials => credentials, :root_path => 'direct-messages/direct_message' }
        internal_request options, &callback
      end

      def self.sent(credentials, &callback)
        options = { :url => Constants::Urls::SENT_MESSAGES, :credentials => credentials, :root_path => 'direct-messages/direct_message' }
        internal_request options, &callback
      end

      def self.delete(message, credentials, &callback)
        request = WebClient.new "#{Constants::Urls::DESTROY_DIRECT_MESSAGE}#{message}#{Constants::RESPONSE_FORMAT}", credentials
        request.get
      end

      def self.send(user, message, credentials, &callback)
        request = WebClient.new Constants::Urls::UPDATE_STATUS, credentials
        params = { :status => message, :user => user, :source => "Sylvester" }
        request.post_and_return params, nil, &callback
      end

      def to_status
        stat = Status.new
        stat.created_at = self.created_at
        stat.user = self.sender
        stat.text = self.text
        stat.id = self.id

        stat
      end





    end

  end
end